docs/move/move-on-aptos/objects/configuring-objects.md

```diff
@@ -0,0 +1,379 @@
+    let linear_transfer_ref = object::generate_linear_transfer_ref(
+      &transfer_ref
+    );
+
+    // Store it for later usage
+    let object_controller = borrow_global_mut<ObjectController>(
+      object_address
+    );
+    option::fill(
+      &mut object_controller.linear_transfer_ref,
+      linear_transfer_ref
+    )
+  }
+
+  /// Now only owner can transfer exactly once
+  entry fun transfer(
+    caller: &signer,
+    object: Object<ObjectController>,
+    new_owner: address
+  ) {
+    let object_address = object::object_address(object);
+
+    // Retrieve the linear_transfer ref, it is consumed so it must be extracted
+    // from the resource
+    let object_controller = borrow_global_mut<ObjectController>(
+      object_address
+    );
+    let linear_transfer_ref = option::extract(
+      &mut object_controller.linear_transfer_ref
+    );
+
+    object::transfer_with_ref(linear_transfer_ref, new_owner);
+  }
+}
+```
+
+## Allowing deletion of an Object
+
+Deleting an object can be useful to get rid of clutter, as well as retrieve back
+storage refunds. Deletion can be done with a `DeleteRef`, which must be created
+at object creation time.
+
+Note that you cannot create a `DeleteRef` for a non-deletable object.
+
+```move
+module my_addr::object_playground {
+  use std::signer;
+  use std::option;
+  use std::string::{self, String};
+  use aptos_framework::object::{self, Object};
+
+  /// Caller is not the owner of the object
+  const E_NOT_OWNER: u64 = 1;
+
+  #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+  struct ObjectController has key {
+    delete_ref: object::DeleteRef,
+  }
+
+  entry fun create_my_object(
+    caller: &signer,
+    transferrable: bool,
+    controllable: bool
+  ) {
+    let caller_address = signer::address_of(caller);
+
+    // Creates the object
+    let constructor_ref = object::create_object(caller_address);
+
+    // Retrieves a signer for the object
+    let object_signer = object::generate_signer(&constructor_ref);
+
+    // Creates and store the delete ref
+    let delete_ref = object::generate_delete_ref(&constructor_ref);
+    move_to(&object_signer, ObjectController {
+      delete_ref
+    });
+    // ...
+  }
+
+  /// Now only let the owner delete the object
+  entry fun delete(
+    caller: &signer,
+    object: Object<ObjectController>,
+  ) {
+    // Only let caller delete
+    let caller_address = signer::address_of(caller);
+    assert!(object::is_owner(&object, caller_address), E_NOT_OWNER);
+
+    let object_address = object::object_address(object);
+
+    // Retrieve the delete ref, it is consumed so it must be extracted
+    // from the resource
+    let ObjectController {
+      delete_ref
+    } = move_from<ObjectController>(
+      object_address
+    );
+
+    // Delete the object forever!
+    object::delete(delete_ref);
+  }
+}
+```
+
+## Immutability
+
+An object can be made immutable by making the contract associated immutable, and
```

Comment:
What if the immutable contract deployed contains functions which mutate the object passed in? i.e.
```
public fun hero_equip_weapon(owner: &signer, hero: Object<Hero>, weapon: Object<Weapon>) acquires Hero {
    let hero_obj = borrow_global_mut<Hero>(object::object_address(&hero));
    option::fill(&mut hero_obj.weapon, weapon);
    object::transfer_to_object(owner, weapon, hero);
}
```

Maybe mention that the contract should contain no functions to mutate objects as well.